config/foreman-proxy-content-answers.yaml

```diff
@@ -13,6 +13,7 @@ certs:
   generate: false
 foreman_proxy_content:
   pulpcore_mirror: true
+  puppet: false
```

Comment:
I've been thinking about this. It's this piece:
https://github.com/theforeman/puppet-foreman_proxy_content/blob/37a6e9eaa058fefd1788625ad5def7243cb351e5/manifests/init.pp#L319-L330

Perhaps we can make it `undef` by default and do:
```puppet
if pick($puppet, 'puppet' in lookup('classes', Array, undef, []) {
}
```

This would rely on Kafo presenting `classes` in Hiera to determine it dynamically. For users outside of Kafo it's still possible to explicit set it.

Reply:
I wondered if there was a way to do roughly the following in a hiera layer:

```
certs::puppet::hostname: "%{lookup($certs::foreman_proxy::hostname)}"
certs::puppet::generate: "%{lookup($puppet::server)}"
certs::puppet::deploy: "%{lookup($puppet::server)}"
```

And then use the order configuration to have:

```
:order:
  - certs
  - foreman
  - katello
  - foreman_proxy
  - foreman_proxy::plugin::pulp
  - foreman_proxy_content
  - certs::puppet
  - puppet
```

Reply:
https://github.com/theforeman/puppet-foreman_proxy_content/pull/383

As for your suggestion, I don't know if that would work. The integration is REALLY hacky now. Perhaps we should move it here instead:
https://github.com/theforeman/puppet-puppet/blob/7534cbe648d6576509303da7be4f7724ee2d1fa4/manifests/server/config.pp#L299-L315

That could get a `katello` boolean and include the class if needed. Let's see about that.